src/appshell/view/preferences/preferencesmodel.cpp

```diff
@@ -198,8 +199,11 @@ void PreferencesModel::load(const QString& currentPageId)
 void PreferencesModel::resetFactorySettings()
 {
     static constexpr bool KEEP_DEFAULT_SETTINGS = true;
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+    QApplication::processEvents();
     configuration()->revertToFactorySettings(KEEP_DEFAULT_SETTINGS);
     configuration()->startEditSettings();
+    QApplication::restoreOverrideCursor();
```

Comment:
not strictly relevant to the crash but it always felt like it was hanging without this